flow/flow.go

```diff
@@ -428,7 +423,11 @@ func (d *Flow) updateFromLogs(containerName string) error {
 		}
 
 		m := map[string]interface{}{}
-		if err := json.Unmarshal(got, &m); err != nil {
+		// This assumes the container is not using a TTY. In this case
+		// stdout/stderr are multiplexed on the same stream and 8-byte header
+		// precedes each line. We don't need to parse the header since we are
+		// using scanner.Bytes().
+		if err := json.Unmarshal(got[8:], &m); err != nil {
```

Comment:
Let's add a length check to not cause an index out of bounds panic in case of something unexpected.

Reply:
Addressed in 7a9bb63